libs/books/data-access/src/lib/+state/reading-list.effects.ts

```diff
@@ -15,9 +15,10 @@ export class ReadingListEffects implements OnInitEffects {
         this.http
           .get<ReadingListItem[]>('/api/reading-list')
           .pipe(
-            map(data =>
-              ReadingListActions.loadReadingListSuccess({ list: data })
-            )
+            map(data => {
+              const bookData = data.filter(b => b.bookId);
```

Comment:
What are you actually doing here?

Why are you filtering out books without a `bookId`? Why would they be returned from the API like this? Is is something you've seen?

Reply:
Getting empty objects in the response of reading-list api(/api/reading-list) occasionally. So I'm doing filtering here to pass only the objects with book items. Attached screenshot below.

<img width="1680" alt="Screen Shot 2020-09-21 at 9 52 00 AM" src="https://user-images.githubusercontent.com/26583729/93843347-2de07980-fc5f-11ea-9bbb-8573d5004bb1.png">

Reply:
I think this indicates that you've been doing some experimenting with the API - there should never be any empty objects in that list.

If this were a real risk, do you really think that filtering in the front end is the right solution?

Reply:
Seems when my code was not working, I have added empty objects. I have removed them at the api level. Now frontend does not need to make the check.

<img width="1680" alt="Screen Shot 2020-09-22 at 1 02 40 PM" src="https://user-images.githubusercontent.com/26583729/93924185-05499580-fcda-11ea-8329-c916af9e0ecf.png">